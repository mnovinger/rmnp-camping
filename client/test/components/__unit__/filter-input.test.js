import renderer from 'react-test-renderer';
import React from 'react';
import sinon from 'sinon';
import { shallow, mount, render } from 'enzyme';

import FilterInput from '../../../src/app/components/filter-input';

describe('The CalendarControl Component', () => {
  it('should render correctly', () => {
    const props = {
      filterTextFt: () => {},
      filterText: "Some Text",
    };

    const tree = renderer.create(<FilterInput { ...props }/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should invoke the callback when clicked', () => {
    var callback = sinon.spy();
    const props = {
      filterTextFt: callback,
      filterText: "Some Text",
    };

    const filterCmp = shallow(<FilterInput { ...props }/>);
    filterCmp.find("input").simulate('change', {target:{value: "a"}});
    expect(callback.called).toBe(true);
  })

});