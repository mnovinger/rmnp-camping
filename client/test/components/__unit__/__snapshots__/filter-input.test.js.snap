// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CalendarControl Component should render correctly 1`] = `
<div
  className="filter-input"
>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="formControlsText"
    >
      Type to Filter
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="site name."
      type="text"
      value="Some Text"
    />
  </div>
</div>
`;
